### PR TITLE
Add ability to list and delete SMS messages in client

### PR DESCRIPTION
**Pull request by dameikle**
Updated Client class to support DELETE requests and provide additional methods to support interacting with messages.

### CODE REVIEW

File: lib/messagebird/client.rb [hunk 1]
```diff
@@ -51,16 +53,16 @@ def request(method, path, params={})
...
       case response.code.to_i
       when 200, 201, 204, 401, 404, 405, 422
-        json = JSON.parse(response.body)
+        unless response.body.nil?
+          json = JSON.parse(response.body)
+          # If the request returned errors, create Error objects and raise.
+          if json.has_key?('errors')
+            raise ErrorException, json['errors'].map { |e| Error.new(e) }
+          end
+        end
```
Comment (marcelcorso) [inline lib/messagebird/client.rb]: if the body is null the `json` wont be defined when we return. Maybe we need to give it a default value.

File: lib/messagebird/client.rb [hunk 2]
```diff
@@ -106,6 +108,18 @@ def otp_verify(recipient, token, params={})
         OTP.new(request(:get, path))
     end
 
+    # Retrieve all messages. Provides the latest message first, unless
+    # ordered_by_received is set to true
+    # Params:
+    # +ordered_by_received+:: boolean - order by received date. Default false.
```
Comment (marcelcorso) [inline lib/messagebird/client.rb]: I don't think `ordered_by_received` is a good idea. The user of that method that should be responsible for reversing the order if they think its a good idea. Better to keep things simple.

**Comment by samwierema** [comment]
@dameikle thanks again for submitting a pull request! We'd love to know if you have questions or feedback based on @marcelcorso's review.

**Comment by Funfun** [comment]
@marcelcorso ^^

**Comment by Funfun** [comment]
closing PR due to long inactivity

**Pull request closed without merging**